src/structure/viewer.ts

```diff
@@ -847,16 +882,111 @@ export class MoleculeViewer {
         this._highlighted.model.setStyle({}, this._mainStyle());
     }
 
+    private _addShapes(): void {
+        if (this._current === undefined) {
+            return;
+        }
+        if (this._options.shape.value === '') {
+            return;
+        }
+
+        this._viewer.removeAllShapes();
+
+        if (this._options.unitCell.value) {
```

Comment:
Why is this here? Is this removed by the `removeAllShapes` above?

Reply:
exactamundo